pkg/wakaama/contrib/objects/on_off_switch.c

```diff
@@ -0,0 +1,492 @@
+                    instance->counter++;
+                }
+            }
+
+            _mark_resource_changed(instance_id, LWM2M_ON_OFF_SWITCH_DIGITAL_INPUT_STATE_ID);
+            break;
+        }
+
+        case LWM2M_ON_OFF_SWITCH_ON_TIME_ID:
+        {
+            int64_t val;
+            lwm2m_data_decode_int(&data_array[i], &val);
+            if (val != 0) {
+                DEBUG("[lwm2m:on_off_switch:write]: invalid on_time value, only can write 0\n");
+                result = COAP_400_BAD_REQUEST;
+            } else {
+                ztimer_stopwatch_reset(&instance->stopwatch);
+                _mark_resource_changed(instance_id, LWM2M_ON_OFF_SWITCH_ON_TIME_ID);
+            }
+            break;
+        }
+
+        case LWM2M_ON_OFF_SWITCH_OFF_TIME_ID:
+        {
+            int64_t val;
+            lwm2m_data_decode_int(&data_array[i], &val);
+            if (val != 0) {
+                DEBUG("[lwm2m:on_off_switch:write]: invalid off_time value, only can write 0\n");
+                result = COAP_400_BAD_REQUEST;
+            } else {
+                ztimer_stopwatch_reset(&instance->stopwatch);
+                _mark_resource_changed(instance_id, LWM2M_ON_OFF_SWITCH_OFF_TIME_ID);
+            }
+            break;
+        }
+
+        case LWM2M_ON_OFF_SWITCH_APP_TYPE_ID:
+            if (data_array[i].type != LWM2M_TYPE_STRING && data_array[i].type != LWM2M_TYPE_OPAQUE) {
+                DEBUG("[lwm2m:on_off_switch:write]: invalid type for app_type"
+                      "(%" PRId8 ")\n", (uint8_t)(data_array[i].type));
+                result = COAP_400_BAD_REQUEST;
+                break;
+            }
+
+            if (data_array[i].value.asBuffer.length >
+                CONFIG_LWM2M_ON_OFF_SWITCH_APP_TYPE_MAX_SIZE - 1) {
+                DEBUG("[lwm2m:on_off_switch:write]: value too big for app_type\n");
+                result = COAP_500_INTERNAL_SERVER_ERROR;
+                break;
+            }
+
+            memcpy(instance->app_type, data_array[i].value.asBuffer.buffer,
+                   data_array[i].value.asBuffer.length);
+            instance->app_type[data_array[i].value.asBuffer.length] = '\0';
+            _mark_resource_changed(instance_id, LWM2M_ON_OFF_SWITCH_APP_TYPE_ID);
+            break;
+        }
+    }
+
+free_out:
+    mutex_unlock(&_on_off_switch_object.lock);
+    return result;
+}
+
+static void _mark_resource_changed(uint16_t instance_id, uint16_t resource_id)
+{
+    lwm2m_uri_t uri;
+    LWM2M_URI_RESET(&uri);
+    uri.objectId = LWM2M_ON_OFF_SWITCH_OBJECT_ID;
+    uri.instanceId = instance_id;
+    uri.resourceId = resource_id;
+
+    lwm2m_client_data_t *client_data;
+    client_data = (lwm2m_client_data_t *)_on_off_switch_object.wakaama_object.userData;
+    lwm2m_resource_value_changed(client_data->lwm2m_ctx, &uri);
+}
+
+
+lwm2m_object_t *lwm2m_object_on_off_switch_init(lwm2m_client_data_t *client_data)
+{
+    /* initialize the instances */
```

Comment:
```suggestion
lwm2m_object_t *lwm2m_object_on_off_switch_init(lwm2m_client_data_t *client_data)
{
    assert(client_data);
    /* initialize the instances */
```
Alternatively, ensure all code can handle `NULL` in the field.

Reply:
Added